siriuspy/siriuspy/devices/timing.py

```diff
@@ -65,22 +73,36 @@ def bucketlist_sync(self):
     @property
     def continuous_state(self):
         """."""
-        return self['ContinuousEvt-Sts']
+        return bool(self['ContinuousEvt-Sts'])
 
     @continuous_state.setter
     def continuous_state(self, value):
         """."""
-        self['ContinuousEvt-Sel'] = bool(value)
+        new_val = bool(value)
+        if self.continuous_state != new_val:
+            self['ContinuousEvt-Sel'] = bool(value)
+
+    @property
+    def state_machine(self):
+        """."""
+        return self['STATEMACHINE']
+
+    @property
+    def state_machine_str(self):
+        """."""
+        return self.StateMachine._fields[self.state_machine]
 
     @property
     def injection_state(self):
         """."""
-        return self['InjectionEvt-Sts']
+        return bool(self['InjectionEvt-Sts'])
 
     @injection_state.setter
     def injection_state(self, value):
         """."""
-        self['InjectionEvt-Sel'] = bool(value)
+        new_val = bool(value)
+        if self.injection_state != new_val:
+            self['InjectionEvt-Sel'] = bool(value)
```

Comment:
I think you can use `self['InjectionEvt-Sel'] = new_val`

Reply:
actually it is better to write integer values to this PV.